### PR TITLE
Kafka key and value as IAST sources

### PR DESCRIPTION
**Pull request by iunanua**
## Motivation

Add two new cases to test kafka keys and values as vulnerability sources

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: tests/appsec/iast/source/test_kafka_key.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@coverage.basic
+@features.iast_source_kafka_key
+@scenarios.integrations
+class TestKafkaKey(BaseSourceTest):
+    """Verify that kafka message key is tainted"""
+
+    endpoint = "/iast/source/kafkakey/test"
+    requests_kwargs = [{"method": "GET"}]
+    source_type = "kafka.message.key"
+    source_names = ["key"]
```
Comment (manuel-alvarez-alvarez) [inline tests/appsec/iast/source/test_kafka_key.py:18]: The name is hardcoded right?

File: tests/appsec/iast/source/test_kafka_value.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@coverage.basic
+@features.iast_source_kafka_value
+@scenarios.integrations
+class TestKafkaValue(BaseSourceTest):
+    """Verify that kafka message value is tainted"""
+
+    endpoint = "/iast/source/kafkavalue/test"
+    requests_kwargs = [{"method": "GET"}]
+    source_type = "kafka.message.value"
+    source_names = ["value"]
```
Comment (manuel-alvarez-alvarez) [inline tests/appsec/iast/source/test_kafka_value.py:18]: Is the source name hardcoded?

File: tests/appsec/iast/source/test_kafka_key.py (hunk 1 above)
Comment (iunanua) [reply tests/appsec/iast/source/test_kafka_key.py]: yes, I have removed them from the tests. As they are hardcoded they don't contribute much

File: tests/appsec/iast/source/test_kafka_value.py (hunk 2 above)
Comment (iunanua) [reply tests/appsec/iast/source/test_kafka_value.py]: removed!

File: utils/_features.py [hunk 3]
```diff
@@ -1229,6 +1229,26 @@ def iast_source_body(test_object):
...
         return test_object
 
+    @staticmethod
+    def iast_source_kafka_key(test_object):
+        """
+        IAST Source: Kafka message key
+
+        https://feature-parity.us1.prod.dog/#/?feature=179
+        """
+        pytest.mark.features(feature_id=179)(test_object)
```
Comment (robertomonteromiguel) [inline utils/_features.py:1239]: why can't I find this feature ID in FPD?

File: utils/_features.py [hunk 4]
```diff
@@ -1229,6 +1229,26 @@ def iast_source_body(test_object):
...
+        return test_object
+
+    @staticmethod
+    def iast_source_kafka_value(test_object):
+        """
+        IAST Source: Kafka message value
+
+        https://feature-parity.us1.prod.dog/#/?feature=180
+        """
+        pytest.mark.features(feature_id=180)(test_object)
```
Comment (robertomonteromiguel) [inline utils/_features.py:1249]: why can't I find this feature ID in FPD?

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]

File: utils/_features.py (hunk 3 above)
Comment (manuel-alvarez-alvarez) [reply utils/_features.py:1239]: Same for me, we should add them manually to the parity dashboard right?

File: utils/_features.py [hunk 5]
```diff
@@ -1229,6 +1229,26 @@ def iast_source_body(test_object):
         pytest.mark.features(feature_id=175)(test_object)
         return test_object
 
+    @staticmethod
+    def iast_source_kafka_key(test_object):
+        """
+        IAST Source: Kafka message key
+
+        https://feature-parity.us1.prod.dog/#/?feature=179
```
Comment (manuel-alvarez-alvarez) [inline utils/_features.py:1237]:
```suggestion
        https://feature-parity.us1.prod.dog/#/?feature=277
```

File: utils/_features.py [hunk 6]
```diff
@@ -1229,6 +1229,26 @@ def iast_source_body(test_object):
...
+        """
+        pytest.mark.features(feature_id=179)(test_object)
+        return test_object
+
+    @staticmethod
+    def iast_source_kafka_value(test_object):
+        """
+        IAST Source: Kafka message value
+
+        https://feature-parity.us1.prod.dog/#/?feature=180
```
Comment (manuel-alvarez-alvarez) [inline utils/_features.py:1247]:
```suggestion
        https://feature-parity.us1.prod.dog/#/?feature=278
```

File: utils/_features.py (hunk 3 above)
Comment (iunanua) [reply utils/_features.py]: hehe, invent total

thank you @manuel-alvarez-alvarez for creating them and show me how to do it!

File: utils/build/docker/nodejs/express4/iast/sources.js [hunk 7]
```diff
@@ -101,6 +114,82 @@ function init (app, tracer) {
...
   })
+
+  app.get('/iast/source/kafkavalue/test', (req, res) => {
+    const kafka = getKafka()
+
+    const producer = kafka.producer()
+    const doKafkaOperations = async () => {
+      await producer.connect()
+      await producer.send({
+        topic: 'iast-system-tests-queue',
```
Comment (manuel-alvarez-alvarez) [inline utils/build/docker/nodejs/express4/iast/sources.js:125]: How are you creating the topic? the only one I could find in system-tests is this [one](https://github.com/DataDog/system-tests/blob/main/utils/_context/containers.py#L591)

Comment (iunanua) [reply utils/build/docker/nodejs/express4/iast/sources.js:125]: ummm, it's created automatically. But if it is a problem we could use `dsm-system-tests-queue`

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
CI lint fail

**Review by manuel-alvarez-alvarez — APPROVED** [review]

**Review by smola — APPROVED** [review]

**Review by jandro996 — APPROVED** [review]

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**